src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt

```diff
@@ -225,54 +239,66 @@ open class HiddenSecretsPlugin : Plugin<Project> {
             dependsOn(TASK_UNZIP_HIDDEN_SECRETS)
 
             doLast {
-                //Copy files if they do not exist
+                //Assert that the key is present
+                getKeyParam()
+                //Copy files if they don't exist
                 copyCppFiles()
-                copyKotlinFiles()
+                copyKotlinFile()
 
                 val keyName = getKeyNameParam()
                 val packageName = getPackageNameParam()
                 val obfuscatedKey = getObfuscatedKey()
 
+                //Add method in Kotlin code
+                var secretsKotlin = getKotlinFile()
+                if (secretsKotlin == null) {
+                    //File not found in project
+                    secretsKotlin = getKotlinDestination(packageName, KOTLIN_FILE_NAME)
+                }
+                if (secretsKotlin.exists()) {
+                    var text = secretsKotlin.readText(Charset.defaultCharset())
+                    text = text.replace(PACKAGE_PLACEHOLDER, packageName)
+                    if (text.contains(keyName)) {
+                        println("⚠️ Method already added in Kotlin !")
+                    }
+                    text = text.dropLast(1)
+                    text += CodeGenerator.getKotlinCode(keyName)
+                    secretsKotlin.writeText(text)
+                } else {
+                    error("Missing Kotlin file, please run gradle task : $TASK_COPY_KOTLIN")
+                }
+                //Resolve package name for Cs++ from the one used in Kotlin file
+                var kotlinPackage = Utils.getKotlinFilePackage(secretsKotlin)
```

Comment:
can be a val ;)

Reply:
@GhassenMsd not it is re-assigned in ligne 274.